### ISSUE TITLE
Display the mini-cart block in the cart and checkout pages when using C&C blocks

### ISSUE THREAD

**Issue by xristos3490** [woothemes/woocommerce#42352]
### Description

Looking at how we manage the mini-cart block, it appears that it is not visible when accessing the cart and checkout pages. This causes a misalignment problem with the primary navigation. See this screenshot:
![](https://cdn-std.droplr.net/files/acc_1208648/yjs4e1)

I recently learned from a conversation with @gigitux that the mini-cart block intentionally remains hidden while shoppers view the cart and checkout pages. This is to prevent any complications that may arise from discrepancies between the cart state and changes made to the cart through legacy cart and checkout shortcodes.

To enhance the user experience, it would be advisable to display the mini-cart exclusively for merchants with active C&C blocks in their stores. This would enable us to synchronize the mini-cart with the global cart state(?)

Hopefully, I'm not missing anything important that prevents us from delivering the described UX. @Aljullu I'd love to know your thoughts on this!

**Comment by Aljullu** [comment]
Thanks for opening this issue @xristos3490!

> Looking at how we manage the mini-cart block, it appears that it is not visible when accessing the cart and checkout pages. This causes a misalignment problem with the primary navigation.

The Mini-Cart block has a setting to change whether it's hidden or completely removed from Cart and Checkout pages, meaning whether it "takes space" or not. Would that work for your use-case?

![imatge](https://github.com/woocommerce/woocommerce-blocks/assets/3616980/edebc623-254b-420e-8f8d-88ddfa8ed39c)

> To enhance the user experience, it would be advisable to display the mini-cart exclusively for merchants with active C&C blocks in their stores. This would enable us to synchronize the mini-cart with the global cart state(?)

This was briefly discussed in https://github.com/woocommerce/woocommerce-blocks/issues/7760, but we decided not to add this option, as we considered the other two covered all valid use-cases. I don't have a strong opinion, but I'm curious, what would be a use-case for showing the Mini-Cart button on the Cart and Checkout pages?

**Comment by xristos3490** [comment]
Thanks for the additional information, @Aljullu! 

> what would be a use-case for showing the Mini-Cart button on the Cart and Checkout pages?

I don't think there's a specific use case for this, honestly. My main concern was centered around the user interface and any inconsistencies happening from hiding this block. 

Some thoughts:
- In my opinion, having a separate toggle to hide or remove this block may be seen as too technical. The terms "hide" and "remove" likely stem from a technical perspective when attempting to reconcile the distinctions between the "visibility" and "display" rules. To a merchant, these two terms may appear synonymous.
- One potential solution to address the UX of the mini-cart in the C&C blocks is to draw some inspiration from the current functionality, where:
  - An interactive mini-cart is displayed on the cart page but is not clickable, and;
  - An interactive mini-cart is displayed on the checkout page, and when the shopper clicks on it, they get redirected to the cart page.

That leaves us with a single block setting:
- Show/Hide the mini-cart on C&C blocks

Both options are quite clear to the merchant, and the "Hide" option, uses a `display` fix to remove any footprint from the DOM.

I have some concerns regarding the mini-cart in the cart being labeled as "non-working" simply because it is not clickable. However, we can work on improving this aspect. Perhaps, something like a "disabled" visual state could work, as you previously suggested.

I'm thinking aloud; don't have a strong opinion, but I believe we could make the UX better.

**Note**
I am talking about a dynamic version of the mini-cart that will appear during checkout. Our goal is to offer last-minute suggestions, such as "add $5 more to get free shipping", to customers at this stage. As a result, changes to the cart may occur during the checkout process.

**Comment by Aljullu** [comment]
Makes sense, thanks for expanding on this, @xristos3490!

I'm going to add the `needs design` label here, as we probably need some design direction before going ahead with the changes.

**Comment by Aljullu** [comment]
It looks like we received another report for this in https://github.com/woocommerce/woocommerce-blocks/issues/9740. @shaunandrews maybe that's something that might be worth prioritizing if you have some time? tl;dr: we currently offer the option to hide or remove the Mini-Cart block button in Cart and Checkout pages, however, there are some use-cases were it might be interesting to display it normally. What do you think?

**Comment by webaxones** [comment]
Hello,
In my opinion, this setting is missing a "Display" option.
I will have to tinker with CSS and JS to show the button, and when we are on the "checkout" page, redirect to the "cart" page.
Because that's how a customer thinks:
I'm on a product page, I click on the cart menu, and it opens the mini-cart, allowing me to see what I can do right away.
I'm on the checkout page, and if I want to go back to the cart, well, I click on the cart icon because it's obvious! A regular customer doesn't think, "wait, on the product page, it opens a mini-cart on the sidebar. But on the checkout..." They think cart icon > cart, whether it's mini or maxi.
So, in my opinion, I would have added a "Display" option (description: redirect to the cart page) or allowed reactivation of the click event programmatically to make it work as desired.
And for me, the option to hide the cart menu on the cart/checkout page is a UX mistake: when I navigate to another page on the website, I don't make its menu disappear. The main menu is, among other things, the element that allows the user to always maintain control of their navigation. It's a compass. And on a compass, you don't remove an index. Creating a visual dissonance or inconsistency will inevitably confuse them.
Thanks,
:)

**Comment by scottlucas** [comment]
We would like the function to show this too as it prevents our customers from returning to their cart easily during checkout, without using the browser back navigation

**Comment by thuautp** [comment]
also requested in 7768732-zen

**Comment by pierorocca** [comment]
Reviewed and confirmed this is still valid. Marking as ready for engineering.

Current evidence: Trunk still explicitly suppresses the Mini Cart frontend script/data on cart and checkout pages, and the issue discussion already converged on design direction as the missing input. This is a design/product decision, not an evidence gap.

Yoann has designs from 2025 flagging altering the options so that hiding -> remove so that the element isn't causing the visible misalignment issues. Let's reference those designs in the Payments Design P2 when addressing this issue.

**Comment by yjailin** [comment]
Coming at this from design.

<img width="277" height="806" alt="Image" src="https://github.com/user-attachments/assets/77147edf-a26a-45c0-9b59-05c298021c7e" />

I think this setting shouldn't exist. A good default would be:

- Never show a cart icon (or its empty space) in the header when browsing the cart page. It would only either open the mini cart over the cart, which is bad, or refresh the page.
- Always show the cart icon in the checkout header, to get back to the cart.

The setting itself is the problem here, not just its states. cc @pierorocca  @elizaan36

**Comment by elizaan36** [comment]
Yes I think we can set smarter defaults here. It feels like this setting passes on unnecessary complexity to merchants.

**Comment by tyxla** [comment]
Also see #55465 as yet another request for the same enhancement.
